tests/Process/ProcessManagerTest.php

```diff
@@ -51,4 +51,51 @@ public function testRunGetErrorOutput(): void
         }
         $this->assertTrue(false, 'Exception should be thrown');
     }
+
+    public function testAddOutputCallback(): void
+    {
+        $processManager = new ProcessManager();
+        $processManager->addOutputCallback(function () {
+            return true;
+        });
+        $property = (new \ReflectionClass($processManager))->getProperty('outputCallbacks');
+        $property->setAccessible(true);
+        foreach ($property->getValue($processManager) as $callback) {
+            $this->assertTrue($callback());
+        }
+    }
+
+    public function testAddStateChangeCallback(): void
+    {
+        $processManager = new ProcessManager();
+        $processManager->addStateChangeCallback(function () {
+            return true;
+        });
+        $property = (new \ReflectionClass($processManager))->getProperty('stateChangeCallbacks');
+        $property->setAccessible(true);
+        foreach ($property->getValue($processManager) as $callback) {
+            $this->assertTrue($callback());
```

Comment:
assertTrue is not conclusively if it tested the StateChangeCallback.
maybe use a big random number, or test the callback itself?